servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletContext.java

```diff
@@ -0,0 +1,591 @@
+            new HashSet<>(Arrays.asList(SessionTrackingMode.COOKIE,SessionTrackingMode.URL));
+    private MimeMappings mimeMappings = new MimeMappings();
+    private final UrlMapper<ServletRegistration> servletUrlMapper = new UrlMapper<>(true);
+    private final UrlMapper<FilterRegistration> filterUrlMapper = new UrlMapper<>(false);
+    private Set<SessionTrackingMode> sessionTrackingModeSet;
+    private String requestCharacterEncoding;
+    private String responseCharacterEncoding;
+    private InetSocketAddress serverAddress;
+    private final String contextPath;
+    private final String servletContextName;
+    private boolean started;
+
+    /**
+     * Constructor.
+     */
+    public DefaultServletContext(String contextPath) {
+        requireNonNull(contextPath, "contextPath");
+        this.contextPath = contextPath;
+        this.servletContextName = contextPath.replace("/","");
+    }
+
+    /**
+     * Is server started.
+     */
+    public boolean isStarted() {
+        return started;
+    }
+
+    /**
+     * Set server started.
+     */
+    public void setStarted(boolean started) {
+        this.started = started;
+    }
+
+    /**
+     * Add a new mime mapping.
+     */
+    public void addMimeMapping(MimeMappings mimeMappings) {
+        requireNonNull(mimeMappings, "mimeMappings");
+        this.mimeMappings  = mimeMappings;
+    }
+
+    /**
+     * Get server address.
+     */
+    public InetSocketAddress getServerAddress() {
+        return serverAddress;
+    }
+
+    /**
+     * Set server address.
+     */
+    public void setServerAddress(InetSocketAddress serverAddress) {
+        requireNonNull(serverAddress, "serverAddress");
+        this.serverAddress = serverAddress;
+    }
+
+    /**
+     * Get servlet path.
+     */
+    public String getServletPath(String absoluteUri) {
+        requireNonNull(absoluteUri, "absoluteUri");
+        return servletUrlMapper.getServletPath(absoluteUri);
+    }
+
+    /**
+     * Get session timeout.
+     */
+    public int getSessionTimeout() {
+        return sessionTimeout;
+    }
+
+    /**
+     * Set session timeout.
+     */
+    public void setSessionTimeout(int sessionTimeout) {
+        checkArgument(sessionTimeout > 0, "sessionTimeout: %s (expected: > 0)", sessionTimeout);
+        this.sessionTimeout = sessionTimeout;
+    }
+
+    @Override
+    public String getContextPath() {
+        return contextPath;
+    }
+
+    @Override
+    public DefaultServletContext getContext(String uripath) {
+        requireNonNull(uripath, "uripath");
+        return this;
+    }
+
+    @Override
+    public int getMajorVersion() {
+        return 3;
+    }
+
+    @Override
+    public int getMinorVersion() {
+        return 0;
+    }
+
+    @Override
+    public int getEffectiveMajorVersion() {
+        return 3;
+    }
+
+    @Override
+    public int getEffectiveMinorVersion() {
+        return 0;
+    }
+
+    @Override
+    public String getMimeType(String file) {
```

Comment:
I think we shouldn't check file. Because method's target is get mime type (http content-type) from file extension instead of real file such as: `.avi` => `video/x-msvideo` . Web browser will display download logo, view pdf or play music base on mime type.

Reply:
I will add @Nullable